packages/vite/src/node/plugins/asset.ts

```diff
@@ -371,8 +371,15 @@ async function fileToBuiltUrl(
     }
   } else {
     // emit as asset
-    const { search, hash } = parseUrl(id)
-    const postfix = (search || '') + (hash || '')
+    // On Mac or Linux platforms, the path resolution is considered an invalid URL by the new URL and an error will be thrown.
+    // So it will be converted into a valid URL.
+    const validUrl = 'http://example.com/' + id.split('/').slice(-1)[0]
+    let { search, hash } = new URL(validUrl)
```

Comment:
Can you use 2nd argument of `new URL`? There is something similar on client:

https://github.com/vitejs/vite/blob/08ff23319964903b9f380859c216b10e577ddb6f/packages/vite/src/client/client.ts#L460

Also do you know if importing `node:url` is necessary? I think we can assume global `URL` exists.

Reply:
> Can you use 2nd argument of `new URL`? There is something similar on client:
> 
Thanks for the reminder, this parameter is very useful. 

> Also do you know if importing `node:url` is necessary? I think we can assume global `URL` exists.

I think it is better to use module imports to be consistent with other modules, such as `performance`.

Reply:
nit: Can you use http://vitejs.dev instead of http://example.com? It wouldn't matter but since it already used in `client.ts` :smile:

Reply:
I found that `http://example.com` was also used in the `static.ts` file. 
Should we replace them all or keep it like this for now?
https://github.com/vitejs/vite/blob/08ff23319964903b9f380859c216b10e577ddb6f/packages/vite/src/node/server/middlewares/static.ts#L127

Reply:
Ah, didn't notice :sweat_smile: I think you can update only `asset.ts` to reduce churn.

Reply:
OK, it's replaced.